app/controllers/transition_notes_controller.rb

```diff
@@ -0,0 +1,47 @@
+class TransitionNotesController < ApplicationController
+  before_action :authorize!
+
+  def authorize!
+    student = Student.find(params[:student_id])
+    educator = current_educator
+    raise Exceptions::EducatorNotAuthorized unless educator && educator.is_authorized_for_student(student)
+    raise Exceptions::EducatorNotAuthorized unless educator && educator.labels.include?('k8_counselor')
+  end
+
+  def create
+    transition_note = TransitionNote.new(transition_note_params.merge({
+      educator_id: current_educator.id,
+      recorded_at: Time.now
+    }))
+
+    if transition_note.save
+      render json: { result: 'ok', is_restricted: transition_note.is_restricted }
+    else
+      render json: { errors: transition_note.errors.full_messages }, status: 422
+    end
+  end
+
+  def update
+    transition_note = TransitionNote.find_by_id(params[:id])
```

Comment:
Similar security comment here.  There's a few different student_ids referenced here: `params[:student_id]` the `params[:transition_note][:student_id]` and then the `student_id` that's on the `TransitionNote` record.

To be safe, you could raise if these aren't all the same.

Reply:
Added a guard for this in `authorize!`.

Reply:
Yeah, for sure there are places where we should tighten this in other controllers too.